File: dotnet_backend/src/AIMedical.Api/Models/Entities/Appointment.cs
namespace AIMedical.Api.Models.Entities;

public enum AppointmentStatus
{
    Scheduled,
    Confirmed,
    Waiting,
    InProgress,
    Completed,
    Cancelled,
    NoShow
}

public enum AppointmentType
{
    Regular,
    Emergency,
    FollowUp,
    Telemedicine
}

public class Appointment
{
    public int Id { get; set; }
    public int TenantId { get; set; }
    public int PatientId { get; set; }
    public int DoctorId { get; set; }
    public DateTime ScheduledAt { get; set; }
    public AppointmentType AppointmentType { get; set; } = AppointmentType.Regular;
    public AppointmentStatus Status { get; set; } = AppointmentStatus.Scheduled;
    public int? TokenNumber { get; set; }
    public int? QueuePosition { get; set; }
    public bool IsEmergency { get; set; } = false;
    public string? Notes { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    // Navigation
    public Patient? Patient { get; set; }
    public User? Doctor { get; set; }
}

File: backend/app/models/appointment.py
from datetime import datetime

from sqlalchemy import Column, DateTime, ForeignKey, Integer, String, Text
from sqlalchemy.orm import relationship

from app.database import Base


class Appointment(Base):
    __tablename__ = "appointments"

    id = Column(Integer, primary_key=True, index=True)
    tenant_id = Column(Integer, ForeignKey("tenants.id"), nullable=False, index=True)
    patient_id = Column(Integer, ForeignKey("patients.id"), nullable=False, index=True)
    doctor_id = Column(Integer, ForeignKey("users.id"), nullable=False, index=True)
    scheduled_at = Column(DateTime, nullable=False)
    appointment_type = Column(String(30), default="regular")  # regular/emergency/follow_up/online
    status = Column(String(20), default="scheduled")  # scheduled/in_progress/completed/cancelled/no_show
    token_number = Column(Integer, nullable=True)
    queue_position = Column(Integer, nullable=True)
    notes = Column(Text, nullable=True)
    created_at = Column(DateTime, default=datetime.utcnow)

    tenant = relationship("Tenant", back_populates="appointments")
    patient = relationship("Patient", back_populates="appointments")
    doctor = relationship("User", foreign_keys=[doctor_id])


class Queue(Base):
    __tablename__ = "queues"

    id = Column(Integer, primary_key=True, index=True)
    tenant_id = Column(Integer, ForeignKey("tenants.id"), nullable=False, index=True)
    date = Column(DateTime, nullable=False)
    doctor_id = Column(Integer, ForeignKey("users.id"), nullable=False, index=True)
    current_token = Column(Integer, default=0)
    total_tokens = Column(Integer, default=0)

    doctor = relationship("User", foreign_keys=[doctor_id])
